### DISCUSSION TITLE
How to debug LDAP authentication

### DISCUSSION THREAD

**Discussion by bahner** [netbox-community/netbox-docker#454] [Getting Help] [answered]
I have add the environment variables as I believe they should be and changed to using ldap-images, but there is nothing in the logs that suggests anything LDAP is taking place.  Also there are no errors logged when checking the logs for either netbox_netbox or netbox_netbox-worker.

The service works fine otherwise, using the admin password.

**Comment by mpibpc-mroose** [comment, marked as answer]
What I do is to implement a custom logging:

```
# logging configuration from scratch
LOGLEVEL = environ.get('LOGLEVEL', 'INFO')

LOGGING = {
    'version': 1,
    'disable_existing_loggers': False,
    'formatters': {
        'verbose': {
            'format': '{levelname} {asctime} {module} {process:d} {thread:d} {message}',
            'style': '{',
        },
        'simple': {
            'format': '{levelname} {message}',
            'style': '{',
        },
    },
    'filters': {
        'require_debug_false': {
            '()': 'django.utils.log.RequireDebugFalse',
        },
    },
    'handlers': {
        'console': {
            'level': LOGLEVEL,
            'filters': ['require_debug_false'],
            'class': 'logging.StreamHandler',
            'formatter': 'simple'
        },
        'mail_admins': {
            'level': 'ERROR',
            'class': 'django.utils.log.AdminEmailHandler',
            'filters': ['require_debug_false']
        }
    },
    'loggers': {
        'django': {
            'handlers': ['console'],
            'propagate': True,
        },
        'django.request': {
            'handlers': ['mail_admins'],
            'level': 'ERROR',
            'propagate': False,
        },
        'django_auth_ldap': {
            'handlers': ['console',],
            'level': LOGLEVEL,
        }
    }
}
```

Then LOGLEVEL in the `netbox.env` can be set to `DEBUG` allowing a deep view into the LDAP debug messages. This could be placed in an `logging_conf.py` in `/configuration`.

**Reply by mpibpc-mroose** [reply]
... and maybe check the `REMOTE_AUTH_BACKEND = 'netbox.authentication.LDAPBackend'`setting from within a running netbox container.

**Reply by ryanmerolle** [reply]
Great write up!

**Reply by bahner** [reply]
I will try that. If it works, I suggest you make this a pull request.

**Reply by ryanmerolle** [reply]
We can just add to the wiki.

**Reply by ryanmerolle** [reply]
It works.  I have an issue for this and plugins extra config https://github.com/netbox-community/netbox-docker/issues/461

**Reply by daniel-zook** [reply]
For those of us not well versed in Python yet, FYI: The top of the .py file for this script requires this line:
`from os import environ`

If that is missing, netbox won't start and the log will show this error: 
`NameError: name 'environ' is not defined`

**Comment by bahner** [comment]
🤗

ons. 3. mar. 2021, 15:12 skrev Marco Roose <notifications@github.com>:

**Comment by HtHNightwolf** [comment]
> For those of us not well versed in Python yet, FYI: The top of the .py file for this script requires this line: `from os import environ`
> 
> If that is missing, netbox won't start and the log will show this error: `NameError: name 'environ' is not defined`

Thanks. Now netbox can start with this configuration. But where (and how) do I see the logs then?Until now I have only worked with
`'filename': '/opt/netbox/django-ldap-debug.log',`
which ended up being a file (obviously).

**Reply by jCaffie** [reply]
Did you ever figure it out?
